src/main/java/COSE/OneKey.java

```diff
@@ -162,6 +189,39 @@ else if (Arrays.equals(alg.get(0).value, ASN1.Oid_rsaEncryption)) {
                 keyMap.Add(KeyKeys.RSA_QI.AsCBOR(), pkdl.get(8).value);
 
                 // todo multi prime keys
+            } else if (Arrays.equals(alg.get(0).value, ASN1.Oid_Ed25519)
```

Comment:
It appears to me that the code for this condition is repeated twice. I imagine it would make sense to extract this to a helper method.

Reply:
Yes, I could extract the condition code out into a separate helper method. I will do that in a coming commit to this PR.

Reply:
I have now extracted this code out into a separate helper method. I put this new method in the ASN1 class as it seems to contextually fit better there. The definitions of the OIDs themselves are in ASN1. 

This change was done in commit 2e79e03030f97d6812e70220a2168847d2aca463